lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -196,26 +173,72 @@ class _Visitor extends SimpleAstVisitor<void> {
   @override
   void visitBinaryExpression(BinaryExpression node) {
     var isDartCoreBoolean = node.staticType?.isDartCoreBool ?? false;
-    if (!isDartCoreBoolean || !_isEqualityTest(node.operator)) {
+    if (!isDartCoreBoolean || !node.operator.isEqualityTest) {
       return;
     }
 
-    if (_hasNonComparableOperands(typeSystem, node)) {
-      rule.reportLint(node);
+    var leftOperand = node.leftOperand;
+    if (leftOperand is NullLiteral) return;
+    var rightOperand = node.rightOperand;
+    if (rightOperand is NullLiteral) return;
+    var leftType = leftOperand.staticType;
+    if (leftType == null) return;
+    var rightType = rightOperand.staticType;
+    if (rightType == null) return;
+
+    if (_nonComparable(leftType, rightType)) {
+      rule.reportLint(
+        node,
```

Comment:
Would it be better to highlight the operator, to reduce the highlight range, or would that be confusing?

Reply:
It would be nice to be smaller. At first I thought the right operand would be a good choice but in `a == b`, the "wrong" value (that the dev will likely change to fix the issue) could be either `a` or `b` (or both, or scrap the whole expression). The operator works, to keep the highlight range small.